### DISCUSSION TITLE
options for changing the value of an input bevor validation

### DISCUSSION THREAD

**Discussion by BennyAlex** [lookfirst/mui-rff#1006] [Q&A]
Hello,

it would be nice if we could modify and value before its gets validated, for example on a email - TextField, calling trim() to remove whitespaces.

If this is already possible, please let me know the solution :)

**Comment by lookfirst** [comment]
Thanks for the question, I moved this to a discussion instead of an issue. If it becomes a real issue, we can open a new one.

I haven't needed it myself, but I'm pretty sure that this functionality is done through RFF. I'd start looking over there first. This library is really just a glue layer between RFF and MUI.

If it turns out we are doing something wrong over here, let me know! I'm also interested in hearing what you figure out. =)

**Reply by BennyAlex** [reply]
So it looks like setting the input type to Email allows white spaces or even trims those.
On MUI, you can set the value manually and therefore change it with the onChange event, but with your libary, we dont normally Pass any values and onChange callbacks to the fields, so Idk how it would be done
